### PR TITLE
chore: reintegrate policy logic into core

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Moving policy resolving into the strapi policy registry

### Why is it needed?

removing all strapi app deps from strapi utils

### CODE REVIEW

File: packages/plugins/graphql/server/src/services/content-api/policy.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ const getPoliciesConfig = propOr([], 'policies');
 
 const createPoliciesMiddleware = (resolverConfig: any, { strapi }: { strapi: Core.Strapi }) => {
   const resolverPolicies = getPoliciesConfig(resolverConfig);
-  const policies = policyUtils.resolve(resolverPolicies, {});
+  const policies = strapi.get('policies').resolve(resolverPolicies, {});
```
Comment (Convly) [inline packages/plugins/graphql/server/src/services/content-api/policy.ts:12]: Don't you need to remove the import too? Or is it used elsewhere?

File: packages/core/utils/src/policy.ts [hunk 2]
```diff
@@ -152,10 +38,4 @@ const createPolicyContext = (type: string, ctx: object) => {
   );
 };
 
-export {
```
Comment (Convly) [inline packages/core/utils/src/policy.ts:155]: Why do we keep some policy-related utils in there?

File: packages/core/core/src/registries/policies.ts [hunk 3]
```diff
@@ -1,75 +1,152 @@
...
+const API_PREFIX = 'api::';
+
+interface PolicyInfo {
+  name: string;
+  config: unknown;
+}
+
+type PolicyConfig = string | PolicyInfo;
+
+interface NamesapceInfo {
```
Comment (Convly) [inline packages/core/core/src/registries/policies.ts:15]:
```suggestion
interface NamespaceInfo {
```

File: packages/plugins/graphql/server/src/services/content-api/policy.ts (hunk 1 above)
Comment (alexandrebodin) [reply packages/plugins/graphql/server/src/services/content-api/policy.ts:12]: used for the policyContext creation which is not part of the registry

File: packages/core/utils/src/policy.ts (hunk 2 above)
Comment (alexandrebodin) [reply packages/core/utils/src/policy.ts:155]: eventually `import policyDomain from '@strapi/core/policy'` but for now I don't want to put everything in the core. not the goal of the PR

**Review by Convly — APPROVED** [review]

**Pull request merged**